lib/analyzer.rb

```diff
@@ -0,0 +1,52 @@
+class Analyzer
+
+    def initialize
+        @vehicle_models = []
+        @most_popular_vehicle = [] 
+    end
+
+    def vehicle_models(vehicles)        
+        vehicles.each do |vehicle|
+            @vehicle_models << vehicle.model
+        end
+        @vehicle_models
+    end
+
+    def model_counter(vehicles)
+        vehicle_models(vehicles)
+        @vehicle_models.tally
+    end
+
+    def most_popular_model(vehicles)
+        most_popular_model = model_counter(vehicles).max_by do |model, count|
+            count
+        end
+        most_popular_model
+    end
+
+    def most_popular_vehicle(vehicles)
+        vehicle_object = vehicles.find do |vehicle|
+            vehicle.model == most_popular_model(vehicles).first
+        end
```

Comment:
I'm not quite sure what this method is doing or returning. Part of the confusion could be in how things are being named and what they're returning. For the `most_popular_model` method, I would expect that method to return a single string listing the most popular model, so then what would `most_popular_vehicle` represent? How is this different than the `most_popular_model` method? I will look at your tests and see if there is documentation there to provide clues! 
I do like your experimentation though!